### ISSUE TITLE
Conda package of rawkit

### ISSUE THREAD

**Issue by SeguinBe** [photoshell/rawkit#116]
Hi,

For some reasons, I had to package libraw in a conda package. The libraw conda recipe is [here](https://github.com/Atanahel/conda-libraw) and the rawkit one is [here](https://github.com/Atanahel/conda-rawkit).

The main reason for the rawkit conda package (instead of just using pip) is that `c_types.util.find_library` doesn't look in many locations to find libraries. Hence when the library is installed in the anaconda library folder it doesn't find it. I had to make a [patch](https://github.com/Atanahel/conda-rawkit/blob/master/patch_binding.patch) for it.

I don't know if that is a correction you would like to add to the main branch.

Anyway long story short, if you're stuck on Ubuntu 14.04 or do not have administrative rights on the computer you're using, you can just use `conda install -c atanahel rawkit=0.5.0` (from [here](https://anaconda.org/Atanahel/rawkit)) to get libraw and rawkit running directly.

**Comment by campaul** [comment]
@Atanahel thank for the report. We want discovery of libraw to be as robust as possible so I'll look into fixing this and other possible edge cases.

**Comment by wadawson** [comment]
Note that this patch seems to work nicely for conda and virtualenv installs when the LibRaw library is located in the "virtualenv"/lib/ directory, however it won't generically work if it is located in a different directory. While less self-contained, I might prefer a more traditional solution that depends the user defined `LD_LIBRARY_PATH=path/list/to/user/lib/directories`.

For example one could change line 23 of your [patch](https://github.com/Atanahel/conda-rawkit/blob/master/patch_binding.patch#L23) to:
`libraw = os.path.join(os.environ['LD_LIBRARY_PATH'], 'libraw' + shared_lib_ext[platform.system()])`
(I am not sure how this will work on Windows machines.)

Another nice option might be to have a ./configure setup option to specify the LibRaw directory.

**Comment by SamWhited** [comment]
It's my understanding that find_library should use LD_LIBRARY_PATH already (it shells out to `/sbin/ldconfig` and `ld` on Linux at least), so I think that should be covered already. Manually guessing would just be a fallback. I'll try to merge something like this today, sorry for the long delay (we don't actively work on rawkit anymore generally, but are always willing to accept PRs).
